src/components/map/Map.tsx

```diff
@@ -24,24 +29,31 @@ interface IMapProps {
 @inject('mapStore')
 @inject('routeStore')
 @observer
-class Map extends React.Component<IMapProps> {
-    private map: L.Map;
+class LeafletMap extends React.Component<IMapProps, IMapState> {
+    private map: Map | null;
     private routeLayerView: RouteLayerView;
 
     constructor(props: IMapProps) {
         super(props);
+        this.state = {
+            refreshMapKey: 1,
+            isMapFullscreen: false,
+        };
     }
 
     public componentDidMount() {
-        this.initializeMap();
-        autorun(() => this.updateMap());
-        this.routeLayerView = new RouteLayerView(this.map, this.props.sidebarStore);
         autorun(() => this.updateRouteLines());
     }
 
     private updateRouteLines() {
-        this.routeLayerView.drawRouteLines(this.props.routeStore!.routes);
-        this.centerMapToRoutes(this.props.routeStore!.routes);
+        if (this.map) {
+            if (!this.routeLayerView) {
+                this.routeLayerView = new RouteLayerView(
+                    this.map.leafletElement, this.props.sidebarStore);
```

Comment:
Should RouteLayerView just observe the store itself?

Reply:
https://github.com/HSLdevcom/jore-map-ui/blob/master/src/components/map/FullscreenControl.tsx